### ISSUE TITLE
System.NullReferenceException in MiniCover.HitServices.TestMethodUtils.HasPdb

### ISSUE THREAD

**Issue by Hunter21007** [lucaslorentz/minicover#86]
Hi, 

First: Great Job!

Actually i am experiencing a null reference exception in TestMethodUtils.HasPdb
This is strange, because it works in another WebService without errors.
Unfortunately i am not able to see where exactly the nullref is thrown and why

Is there a "simple" ( :) ) way to run the stuff with debug info?

Could this `lambda_method(Closure , IServiceProvider , Object[] )`  be a problem?

```
System.NullReferenceException: Object reference not set to an instance of an object.
         at MiniCover.HitServices.TestMethodUtils.HasPdb(MethodBase methodBase)
         at MiniCover.HitServices.TestMethodUtils.GetTestMethod()
         at MiniCover.HitServices.HitService.MethodContext..ctor(String filePath)
         at MiniCover.HitServices.HitService.EnterMethod(String fileName)
         at Sfb.Comm.MailTransfer.Service.Controllers.HealthController..ctor(ServiceStartUpConfig startUpConf, IOptions`1 mailTransferConfig, IOptions`1 openIdConfig, ILogger`1 logger)
         at lambda_method(Closure , IServiceProvider , Object[] )
         at Microsoft.AspNetCore.Mvc.Controllers.ControllerFactoryProvider.<>c__DisplayClass5_0.<CreateControllerFactory>g__CreateController|0(ControllerContext controllerContext)
         at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
         at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeInnerFilterAsync>d__14.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeFilterPipelineAsync>d__17.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeAsync>d__15.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.<Invoke>d__6.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Sfb.Extensions.Mvc.ExceptionMiddleware.<Invoke>d__7.MoveNext()
```

**Comment by Hunter21007** [comment]
Same occurs in this class

```
using System;
 using Microsoft.Extensions.Logging;
 using Sfb.JobManagement;
 using Microsoft.Extensions.DependencyInjection;
 using Sfb.Comm.MailTransfer.ApiSchema;
 using Sfb.Extensions.Logging;
  
 namespace Sfb.Comm.MailTransfer.Service.JobManagement
 {
     public class MailTransferJobFactory : JobFactoryBase, IJobFactory
     {
ⓘ        public MailTransferJobFactory(IServiceProvider provider, ILogger<MailTransferJobFactory> logger) : base(provider, logger) { }
  
         public IJob GetSendMailJob(string appName, MailMessage message)
         {
             _log.Info("Going to create new SendMail Job for {subject}", message.Subject);
             return _provider.GetService<IJob>()
                 .SetSequence(_provider.GetService<IParallel>()
                     .SetTasks(_provider.GetService<SendMailTask>()
                         .SetData(appName, message)));
         }
     }
 }
```
Coverage report marks constructor line as visited. Could there be a problem with inheritance? (JobFactoryBase is from a nuget package without pdbs)

```
System.NullReferenceException: Object reference not set to an instance of an object.
         at MiniCover.HitServices.TestMethodUtils.HasPdb(MethodBase methodBase)
         at MiniCover.HitServices.TestMethodUtils.GetTestMethod()
         at MiniCover.HitServices.HitService.MethodContext..ctor(String filePath)
         at MiniCover.HitServices.HitService.EnterMethod(String fileName)
         at Sfb.Comm.MailTransfer.Service.JobManagement.MailTransferJobFactory..ctor(IServiceProvider provider, ILogger`1 logger)
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProvider provider)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitTransient(TransientCallSite transientCallSite, ServiceProvider provider)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, ServiceProvider provider)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitTransient(TransientCallSite transientCallSite, ServiceProvider provider)
         at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(IServiceCallSite callSite, TArgument argument)
         at Microsoft.Extensions.DependencyInjection.ServiceProvider.<>c__DisplayClass22_0.<RealizeService>b__0(ServiceProvider provider)
         at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
         at Microsoft.Extensions.Internal.ActivatorUtilities.GetService(IServiceProvider sp, Type type, Type requiredBy, Boolean isDefaultParameterRequired)
         at lambda_method(Closure , IServiceProvider , Object[] )
         at Microsoft.AspNetCore.Mvc.Controllers.ControllerFactoryProvider.<>c__DisplayClass5_0.<CreateControllerFactory>g__CreateController|0(ControllerContext controllerContext)
         at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
         at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.<InvokeInnerFilterAsync>d__14.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeFilterPipelineAsync>d__17.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.<InvokeAsync>d__15.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.<Invoke>d__6.MoveNext()
      --- End of stack trace from previous location where exception was thrown ---
         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
         at Sfb.Extensions.Mvc.ExceptionMiddleware.<Invoke>d__7.MoveNext()
```

**Comment by georgepaoli** [comment]
In release **2.0.0-ci-20180315091931** works fine. But last release not working.

**Comment by shiro-appdev** [comment]
Thank you very much for the useful tool!!
I'm trying .net core version 2.0.
MSTest works without errors, but in the case of xUnit testing I encountered the same error.

**Comment by felixfbecker** [comment]
I have the same error, running intrumented C# cmdlets in PowerShell:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at MiniCover.HitServices.TestMethodUtils.HasPdb(MethodBase methodBase)
   at MiniCover.HitServices.TestMethodUtils.GetTestMethod()
   at MiniCover.HitServices.HitService.MethodContext..ctor(String filePath)
   at MiniCover.HitServices.HitService.EnterMethod(String fileName)
   at Kubectl.KubeCmdlet..ctor()
   at Kubectl.KubeApiCmdlet..ctor()
   at Kubectl.Cmdlets.GetKubePodCmdlet..ctor()
   at lambda_method(Closure )
   at System.Management.Automation.CommandProcessor.ConstructInstance(Type type)
   at System.Management.Automation.CommandProcessor.Init(CmdletInfo cmdletInformation)
```

**Comment by felixfbecker** [comment]
Can confirm that **2.0.0-ci-20180315091931** works.

**Comment by patricksadowski** [comment]
The parameter `methodBase` of the method `HasPdb` can point to a lambda method that doesn't have a `DeclaringType`. Source of exception: [TestMethodUtils.HasPdb](https://github.com/lucaslorentz/minicover/blob/7df64d07355e2d5a0573184841149c3fd2b3e025/src/MiniCover.HitServices/TestMethodUtils.cs#L42)

**Comment by JorritSalverda** [comment]
I ran into this as well in my ci/cd pipeline. It turned out to be caused by running tests with the `--no-build` flag; apparently something needs to be recompiled. Hope this helps anyone else running into this issue.

**Comment by lucaslorentz** [comment]
Fixed on version v3.0.0-alpha-1
